include/entity.h

```diff
@@ -2137,40 +2137,40 @@ typedef struct {
 typedef struct {
     /* 0x7C */ struct Entity* parent;
     /* 0x80 */ struct Entity* entity;
-    /* 0x84 */ s16 unk84;
-    /* 0x86 */ s16 : 16;
-    /* 0x88 */ s32 : 32;
-    /* 0x8C */ struct Entity* unk8C;
-    /* 0x90 */ struct Entity* unk90;
+    /* 0x84 */ struct Entity* entity84;
+    /* 0x88 */ struct Entity* entity88;
+    /* 0x8C */ struct Entity* entity8C;
+    /* 0x90 */ struct Entity* entity90;
     /* 0x94 */ s32 velocityX;
     /* 0x98 */ s32 velocityY;
     /* 0x9C */ u8 unk9C;
     /* 0x9D */ u8 unk9D;
     /* 0x9E */ u8 unk9E;
     /* 0x9F */ u8 unk9F;
-    /* 0xA0 */ u32 : 32;
-    /* 0xA4 */ s32 posX;
-    /* 0xA8 */ s32 posY;
-    /* 0xAC */ u8 unkAC;
-    /* 0xAD */ u8 unkAD;
+    /* 0xA0 */ struct Entity* entityA0;
+    /* 0xA4 */ struct Entity* entityA4;
+    /* 0xA8 */ s32 : 32;
+    /* 0xAC */ u8 hitParams;
+    /* 0xAD */ u8 hitParams2;
 } ET_Diplocephalus;
 
 typedef struct {
     /* 0x7C */ struct Entity* parent;
     /* 0x80 */ struct Entity* entity;
-    /* 0x84 */ s16 : 16;
+    /* 0x84 */ s16 unk84;
     /* 0x86 */ s16 : 16;
     /* 0x88 */ s32 : 32;
-    /* 0x8C */ s32 : 32;
+    /* 0x8C */ struct Entity* entity8C;
     /* 0x90 */ struct Entity* unk90;
-    /* 0x94 */ s32 : 32;
+    /* 0x94 */ s32 velocityX;
     /* 0x98 */ s32 velocityY;
-    /* 0x9C */ s16 : 16;
+    /* 0x9C */ u8 unk9C;
+    /* 0x9D */ u8 unk9D;
     /* 0x9E */ u8 unk9E;
-    /* 0x9F */ u8 : 8;
-    /* 0xA0 */ u32 : 32;
-    /* 0xA4 */ s32 posX;
-    /* 0xA8 */ s32 posY;
+    /* 0x9F */ u8 unk9F;
+    /* 0xA0 */ s32 : 32;
+    /* 0xA4 */ s32 unkA4;
+    /* 0xA8 */ s32 unkA8;
     /* 0xAC */ s32 unkAC;
 } ET_DiplocephalusUnk;
```

Comment:
I'm still not 100% certain about these two structs, I have a feeling like `ET_Diplocephalus` is the main entity struct, and `ET_DiplocephalusUnk` is for the subentities like the torso, tail, feet etc.

But will need to study the code further in debugger to fully decode this